src/containers/TransactionPage/TransactionPage.duck.js

```diff
@@ -348,14 +379,19 @@ export const fetchTransaction = (id, txRole) => (dispatch, getState, sdk) => {
     });
 };
 
-export const acceptSale = id => (dispatch, getState, sdk) => {
+export const acceptSale = id => async (dispatch, getState, sdk) => {
   if (acceptOrDeclineInProgress(getState())) {
     return Promise.reject(new Error('Accept or decline already in progress'));
   }
   dispatch(acceptSaleRequest());
 
+  const currentTransactions = await sdk.transactions.show({ id });
+  const transType = currentTransactions.data.data.attributes.lastTransition === TRANSITION_EXPIRE_FULL_REFUND_PERIOD
```

Comment:
Do you really need to check it here? Because you can make use of the `nextTransition` data